packages/flutter_tools/test/android/android_device_test.dart

```diff
@@ -157,6 +158,44 @@ Use the 'android' tool to install them:
       ProcessManager: () => mockProcessManager,
     });
   });
+
+  testUsingContext('isSupportedForProject is true on module project', () async {
+    fs.file('pubspec.yaml')
+      ..createSync()
+      ..writeAsStringSync(r'''
+name: example
+
+flutter:
+  module: {}
+''');
+    fs.file('.packages').createSync();
+    final FlutterProject flutterProject = await FlutterProject.current();
+
+    expect(AndroidDevice('test').isSupportedForProject(flutterProject), true);
```

Comment:
What about the contents of the pubspec.yaml file informs whether the project supports Android? I don't see anything mentioning Android in the pubspec.yaml file.